src/libraries/System.Private.CoreLib/src/System/Environment.cs

```diff
@@ -201,23 +201,7 @@ public static OperatingSystem OSVersion
             }
         }
 
-        public static Version Version
-        {
-            get
-            {
-                string? versionString = typeof(object).Assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>()?.InformationalVersion;
-
-                ReadOnlySpan<char> versionSpan = versionString.AsSpan();
-
-                // Strip optional suffixes
-                int separatorIndex = versionSpan.IndexOfAny('-', '+', ' ');
-                if (separatorIndex >= 0)
-                    versionSpan = versionSpan.Slice(0, separatorIndex);
-
-                // Return zeros rather then failing if the version string fails to parse
-                return Version.TryParse(versionSpan, out Version? version) ? version : new Version();
-            }
-        }
+        public static Version Version => Version.Parse(ProductVersionWithoutLabel);
```

Comment:
This can be source generated like `new Version(9, 0, 0)` - no need to invoke the parser.

Reply:
The generated file now looks like:

```c#
// <auto-generated/>

namespace System;

public static partial class Environment
{
    internal const string ProductVersionWithLabel = "9.0.0-dev";
    private static readonly Version s_productVersion = new Version(9, 0, 0);
}
```

Reply:
> private static readonly Version s_productVersion = new Version(9, 0, 0);

This was not cached before. I would not cache it just because we can. Caching is not free - it will make Environment static constructor more expensive.

Reply:
Maybe we can cut the extra ceremony and the auto-generated file can be:
```
namespace System
{
    public static partial class Environment
    {
        public Version Version => new Version(9, 0, 0);
    }
}

namespace System.Runtime.InteropServices
{
    public static partial class RuntimeInformation
    {
        public static string FrameworkDescription => ".NET 9.0.0-dev";
    }
}
```

Reply:
Done:
```c#
// <auto-generated/>

namespace System
{
    public static partial class Environment
    {
        /// <summary>
        /// Gets a version consisting of the major, minor, build, and revision numbers of the common language runtime.
        /// </summary>
        public static Version Version => new Version(9, 0, 0);
    }
}

namespace System.Runtime.InteropServices
{
    public static partial class RuntimeInformation
    {
        /// <summary>
        /// Gets the name of the .NET installation on which an app is running.
        /// </summary>
        public static string FrameworkDescription => ".NET 9.0.0-dev";
    }
}
```